src/com/palmergames/bukkit/towny/TownySettings.java

```diff
@@ -1839,52 +1843,55 @@ public static boolean isRemovingOnMonarchDeath() {
 	}
 
-    public static double getTownUpkeepCost(Town town) {
-    	 
-        double multiplier;
- 
-        if (town != null) {
-            if (isUpkeepByPlot()) {
-                multiplier = town.getTownBlocks().size(); // town.getTotalBlocks();
-            } else {
-                multiplier = Double.valueOf(getTownLevel(town).get(TownySettings.TownLevel.UPKEEP_MULTIPLIER).toString());
-            }
-        } else
-            multiplier = 1.0;
- 
-        Double amount = 0.0;
-        if (town.hasNation()) {
-            double nationMultiplier = 1.0;
-            try {
-                nationMultiplier = Double.valueOf(getNationLevel(town.getNation()).get(TownySettings.NationLevel.NATION_TOWN_UPKEEP_MULTIPLIER).toString());
-            } catch (NumberFormatException e) {
-                e.printStackTrace();
-            } catch (NotRegisteredException e) {
-                e.printStackTrace();
-            }
-            if (isUpkeepByPlot()) {
-            	if (isTownLevelModifiersAffectingPlotBasedUpkeep())
-            		amount = (((getTownUpkeep() * multiplier) * Double.valueOf(getTownLevel(town).get(TownySettings.TownLevel.UPKEEP_MULTIPLIER).toString())) * nationMultiplier);
-            	else
-            		amount = (getTownUpkeep() * multiplier) * nationMultiplier;
-            	if (TownySettings.getPlotBasedUpkeepMinimumAmount() > 0.0 && amount < TownySettings.getPlotBasedUpkeepMinimumAmount())
-               		amount = TownySettings.getPlotBasedUpkeepMinimumAmount();
-                return amount;
-            } else
-                return (getTownUpkeep() * multiplier) * nationMultiplier;
-        } else {
-            if (isUpkeepByPlot()) {
-            	if (isTownLevelModifiersAffectingPlotBasedUpkeep())
-            		amount = (getTownUpkeep() * multiplier) * Double.valueOf(getTownLevel(town).get(TownySettings.TownLevel.UPKEEP_MULTIPLIER).toString());
-            	else
-            		amount = getTownUpkeep() * multiplier;
-                if (TownySettings.getPlotBasedUpkeepMinimumAmount() > 0.0 && amount < TownySettings.getPlotBasedUpkeepMinimumAmount())
-               		amount = TownySettings.getPlotBasedUpkeepMinimumAmount();
-                return amount;
-            } else
-                return getTownUpkeep() * multiplier;
-        }
-    }
+	public static double getTownUpkeepCost(Town town) {
+		TownUpkeepCalculationEvent event = new TownUpkeepCalculationEvent(town,getTownUpkeepCostRaw(town));
+		Bukkit.getPluginManager().callEvent(event);
+		return event.getUpkeep();
+	}
+
+	private static double getTownUpkeepCostRaw(Town town) {
+		double multiplier;
+
+		if (town != null) {
+			if (isUpkeepByPlot()) {
+				multiplier = town.getTownBlocks().size(); // town.getTotalBlocks();
+			} else {
+				multiplier = Double.parseDouble(getTownLevel(town).get(TownySettings.TownLevel.UPKEEP_MULTIPLIER).toString());
+			}
+		} else
+			multiplier = 1.0;
+
+		double amount;
```

Comment:
This variable is assigned within specific scopes and not used outside of them, so it's best to initialize it within the appropriate scopes instead of here.